packages/debugger/src/handler.ts

```diff
@@ -54,43 +54,25 @@ function updateIconButton(
     tooltip: trans.__('Enable / Disable Debugger'),
     onClick
   });
-  widget.toolbar.addItem('debugger-icon', icon);
+  widget.toolbar.insertBefore('kernelName', 'debugger-icon', icon);
 
   return icon;
 }
 
 /**
- * Add a toggle button to the widget toolbar to enable and disable debugging.
+ * Updates button state to on/off,
+ * adds/removes css class to update styling
  *
- * @param widget The widget to add the debug toolbar button to.
- * @param onClick The callback when the toolbar button is clicked.
+ * @param widget the debug button widget
+ * @param on true if switched on, false otherwise
  */
-function updateToggleButton(
-  widget: DebuggerHandler.SessionWidget[DebuggerHandler.SessionType],
-  sessionStarted: boolean,
-  onClick: () => void,
-  translator?: ITranslator
-): Switch {
-  translator = translator || nullTranslator;
-  const trans = translator.load('jupyterlab');
-
-  const button = new Switch();
-  button.id = 'jp-debugger';
-  button.value = sessionStarted;
-  button.caption = trans.__('Enable / Disable Debugger');
-  button.handleEvent = (event: Event) => {
-    event.preventDefault();
-    switch (event.type) {
-      case 'click':
-        onClick();
-        break;
-      default:
-        break;
-    }
-  };
-
-  widget.toolbar.addItem('debugger-button', button);
-  return button;
+function updateIconButtonState(widget: ToolbarButton, on: boolean) {
+  const cls = 'jp-debug-button-on';
```

Comment:
...or `lm-mod-toggled` which is [already used](https://github.com/jupyterlab/jupyterlab/blob/bae0dd23d2e81cfbaaa9b7d9e67b1ac1383b8a29/packages/apputils/src/toolbar.tsx#L928-L930) for closely related `CommandToolbarButton`s (I am not sure if using the lumino prefix is a good idea, but whatever).

As mentioned in the main commend we also need to add `aria-pressed` state.

In a perfect world the changes would be made to the `ToolbarButton` so that we can:
- read the state from the outside easily
- ensure the correct class name and aria attribute gets set consistently across JupyterLab UI.

Reply:
Thanks for reviewing this change!

IntelliJ uses a more subtle icon with a small dot when it is active, let me know how you feel about this. 
<img width="545" alt="Screen Shot 2021-08-02 at 9 17 33 AM" src="https://user-images.githubusercontent.com/289369/127896324-935ccfe5-c434-4b67-84fa-b0e643377dd1.png">
In any case, can you help with the image/svg for the icon. Thanks!

Agree on the state specific tooltip, let me work on that along with other accessibility & code suggestions.

Reply:
I like the filled circle idea, it somewhat connects to how the filled circle is used as "notebook is running" and may be better than the triangle which could indicate "press me to run".

Reply:
@krassowski good point on the accessibility of the toggle icon. I too like the bug+circle idea inspired by IntelliJ.

Reply:
Thanks both!
@krassowski I am getting help from AWS UXD for the icon.